llvmlite/ir/types.py

```diff
@@ -196,6 +196,39 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(FunctionType)
 
+class VectorType(Type):
+
+    @classmethod
+    def __new__(cls, count, typ):
```

Comment:
There's no need to use `__new__` here.  This should be a `__init__` instead.    Also, a `__new__` doesn't need `@classmethod`.